scripts/compress_images.py

```diff
@@ -0,0 +1,196 @@
+# coding: utf-8
+#
+# Copyright 2025 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script to compress images in the repository using GraphicsMagick."""
+
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+import tempfile
+
+from PIL import Image
+from typing import List, TypedDict, Union
+
+ALL_IMAGES_EXTENSIONS = {'.png', '.jpg', '.jpeg', '.webp'}
+IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION = {'.png'}
+IMAGE_EXTENSIONS_SUPPORTING_LZW_COMPRESSION = {'.jpg', '.jpeg', '.webp'}
+TOLERANCE = 0.99
+OUTPUT_DIR = 'compressed_images'
+
+class CompressedImageInfo(TypedDict):
+    """Type definition for compressed image information."""
+
+    path: pathlib.Path
+    original_size: int
+    new_size: int
+
+
+def get_compressible_images(
+        input_path: Union[str, pathlib.Path]
+    ) -> List[CompressedImageInfo]:
+    """Find images that can be compressed further.
+
+    This function scans a directory for images, attempts to compress them using 
+    GraphicsMagick, and identifies those that can be reduced in size. The 
+    compression benchmark is set to 99% of the original file size, meaning only 
+    images that achieve at least a 1% reduction are considered compressible.
```

Comment:
achieve --> admit

(the compressor is achieving something; the image isn't doing the achieving)